### ISSUE TITLE
Use Caching for Dependencies

### ISSUE THREAD

**Issue by JustinGrote** [Azure/AzOps#383]
Replacing the dependencies section of each Github Action (or using a composite action) with this will shave 1-2 minutes off every execution

```yaml
      #
      # Dependencies
      # Install required runtime modules
      #
      - name: Fetch Latest AzOps Version
        id: AzOpsLatestVersion
        run: |
          $latestVersionUri = "https://www.powershellgallery.com/api/v2/FindPackagesById()?id='AzOps'&`$filter=IsLatestVersion"
          $latestVersionId = (Invoke-RestMethod $latestVersionUri).properties.NormalizedVersion
          "::set-output name=latestVersionDate::$latestVersionId"

      - name: Cache
        uses: actions/cache@v2
        with:
          path: |
            ~/.local/share/powershell/Modules
          key: AzOps-${{steps.AzOpsLatestVersion.outputs.latestVersionDate}}
      - name: "Dependencies"
        run: |
          $ProgressPreference = 'SilentlyContinue'
          if (Get-Module "AzOps" -ListAvailable) {
            "AzOps restored from cache."
            return
          }
          Remove-Item -Force -Recurse $HOME/.local/share/powershell/Modules
          Install-Module -Name "AzOps" -Force -PassThru
 ```

**Comment by daltondhcp** [comment]
Will be solved in #459

**Comment by daltondhcp** [comment]
Closing since this is now implemented in GHA and ADO per https://github.com/Azure/AzOps-Accelerator/pull/108 and https://github.com/Azure/AzOps-Accelerator/pull/103
